### PR TITLE
Flagged prompt data in events

### PR DESCRIPTION
**Pull request by rafax**
This PR:

- extends event metadata with information about flagging reason and prompt prefix (only for .com users)
- drops logging of flagged request metadata (which duplicated the same data as events at 2x storage cost) 

Additionally, request metadata is logged when running locally.

## Test plan

- unit tests
- tested locally

### CODE REVIEW

File: cmd/cody-gateway/internal/httpapi/completions/anthropic.go [hunk 1]
```diff
@@ -108,36 +119,23 @@ func NewAnthropicHandler(
...
+				if result, err := isFlaggedAnthropicRequest(anthropicTokenizer, ar, promptRegexps); err != nil {
 					logger.Error("error checking anthropic request - treating as non-flagged",
 						log.Error(err))
-				} else if flagged {
-					// For now, just log the error, don't modify the request / response
-					promptPrefix := ar.Prompt
-					if len(promptPrefix) > logPromptPrefixLength {
-						promptPrefix = promptPrefix[0:logPromptPrefixLength]
-					}
-					logger.Info(flaggedPromptLogMessage,
```
Comment (rafax) [inline cmd/cody-gateway/internal/httpapi/completions/anthropic.go:125]: FYI @willdollman - I'm dropping the log message (so our log-based metric will stop working, but we have redash queries)

Comment (willdollman) [reply cmd/cody-gateway/internal/httpapi/completions/anthropic.go:125]: Our current manual process relies on being able to extract a list of user IDs from logs explorer. Is there a way to extract this list from redash/bigquery once this is removed?

Comment (eseliger) [reply cmd/cody-gateway/internal/httpapi/completions/anthropic.go:125]: could we keep both in place if that makes things harder for you?

File: cmd/cody-gateway/internal/actor/actor.go [hunk 2]
```diff
@@ -62,6 +62,11 @@ func (a *Actor) GetSource() codygateway.ActorSource {
 	return codygateway.ActorSource(a.Source.Name())
 }
 
+func (a *Actor) IsDotComActor() bool {
+	// Corresponds to sourcegraph.com subscription ID.
+	return a != nil && a.ID != "d3d2b638-d0a2-4539-a099-b36860b09819"
```
Comment (eseliger) [inline cmd/cody-gateway/internal/actor/actor.go:67]: shouldn't it be 
```
	return a != nil && a.ID == "d3d2b638-d0a2-4539-a099-b36860b09819"
```
? Maybe a quick unit test for this pretty critical function would be good.

**Review by eseliger — APPROVED** [review]

File: cmd/cody-gateway/internal/actor/actor.go (hunk 2 above)
Comment (rafax) [reply cmd/cody-gateway/internal/actor/actor.go]: Good catch, I changed this to test locally and submitted by accident. Added unit tests to both checks.

File: cmd/cody-gateway/internal/httpapi/completions/anthropic.go (hunk 1 above)
Comment (rafax) [reply cmd/cody-gateway/internal/httpapi/completions/anthropic.go:125]: We could keep both in place, but I answered @willdollman question in Slack (we have an easier alternative than logs)

**Review by willdollman — APPROVED** [review]

**Pull request merged**